code/game/objects/items/miscellaneous.dm

```diff
@@ -144,6 +144,109 @@
 	new choice(get_turf(M))
 	to_chat(M, "You hear something crackle from the beacon for a moment before a voice speaks.  \"Please stand by for a message from S.E.L.F. Message as follows: <span class='bold'>Item request received. Your package has been transported, use the autosurgeon supplied to apply the upgrade.</span> Message ends.\"")
 
+/obj/item/choice_beacon/magic
+	name = "beacon of summon magic"
+	desc = "Not actually magical."
+
+/obj/item/choice_beacon/magic/generate_display_names()
+	var/static/list/magic_item_list
+	if(!magic_item_list)
+		magic_item_list = list()
+		var/list/templist = typesof(/obj/item/storage/box/magic) //we have to convert type = name to name = type, how lovely!
```

Comment:
Actually couldn't you just loop through the typesof with something like `for(var/atom/A in typesof(whatever)` and then do `magic_item_list[A.name] = A`?

Reply:
this uses the code that the Curator beacon uses, didn't want to bother coding it again from scratch

Reply:
okay fair I guess